source/Halibut/Transport/Protocol/MessageExchangeStream.cs

```diff
@@ -63,27 +64,29 @@ async Task SendIdentityMessageAsync(string identityLine, CancellationToken cance
 
         public async Task SendNextAsync(CancellationToken cancellationToken)
         {
-            SetShortTimeoutsAsync();
-            await SendControlMessageAsync(Next, cancellationToken);
-            SetNormalTimeoutsAsync();
+            await WithTimeout(
+                MessageExchangeStreamTimeout.ControlMessageExchangeShortTimeout,
+                async () => await SendControlMessageAsync(Next, cancellationToken));
         }
 
         public async Task SendProceedAsync(CancellationToken cancellationToken)
         {
-            await SendControlMessageAsync(Proceed, cancellationToken);
+            await WithTimeout(
+                MessageExchangeStreamTimeout.ControlMessageExchangeShortTimeout,
+                async () => await SendControlMessageAsync(Proceed, cancellationToken));
```

Comment:
As we discussed, this is different to what was there before, but we believed that this message does not need the full timeout set, and would make sense to use the lower timeout.

Reply:
Ahh, good spot. I did not mean for this to make it into this PR so will revert!!

Reply:
For a future PR